Memory bloat compiling sqlalchemy with MSVC

# Bug Report

## 🐛 Bug Description

Compiling sqlalchemy with MSVC on Python 3.12 on Windows results in a very high memory usage in access of 11GB. On my system with 16GB of RAM this results in running out of memory in the backend or linking step.

## 🖥️ Environment

### 1. Nuitka Version, Python Version, OS, and Platform

```sh
4.0.1
Commercial: None
Python: 3.12.8 (tags/v3.12.8:2dc476b, Dec  3 2024, 19:30:04) [MSC v.1942 64 bit (AMD64)]
Flavor: CPython Official
Executable: ~\Desktop\New folder (9)\venv-3.12\Scripts\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 11
Version C compiler: cl (cl 14.3).
```

### 2. How Nuitka and Python were Installed

via pip in a venv

### 3. Relevant PyPI Packages and Versions

```sh
SQLAlchemy               2.0.46      C:\Users\Tim\Desktop\New folder (9)\venv-3.12\Lib\site-packages pip
```

```sh
SQLAlchemy 2.0.46 pip
```

## 🛠️ To Reproduce

### 2. Short, Self-Contained, Correct, Eligible (SSCCE) Example

```python
# Compilation instructions
# nuitka-project: --standalone

from sqlalchemy.dialects import oracle
print(oracle)
```

## 📉 Expected Behavior

Compile without running out of memory

## 📄 Actual Behavior & Output

Runs out memory in the backend or linking step:

```
(venv-3.12) PS C:\Users\Tim\Desktop\New folder (9)> python -m nuitka test.py
Nuitka-Options: Used command line options:
Nuitka-Options:   --standalone test.py
Nuitka: Starting Python compilation with:
Nuitka:   Version '4.0.1' on Python 3.12 (flavor 'CPython Official') commercial grade 'not installed'.
Nuitka-Inclusion: Including Windows Runtime DLLs, which increases distribution size. Use
Nuitka-Inclusion: '--include-windows-runtime-dlls=no' to disable, or make explicit with
Nuitka-Inclusion: '--include-windows-runtime-dlls=yes'.
Nuitka: Completed Python level compilation and optimization.
Nuitka: Generating source code for C backend compiler.
Nuitka: Running data composer tool for optimal constant value handling.
Nuitka: Running C compilation via Scons.
Nuitka-Scons: Backend C compiler: cl (cl 14.3).
Nuitka-Scons: Backend C linking with 223 files (no progress information available for this stage).
Unexpected output from this command:
link @C:\Users\Tim\AppData\Local\Temp\tmpvdim422v.lnk
Generating code
C:\Users\Tim\Desktop\New folder (9)\test.build\module.sqlalchemy.dialects.oracle.dictionary.c(2660) : fatal error C1002: compiler is out of heap space in pass 2
LINK : fatal error LNK1257: code generation failed
scons: *** [C:\Users\Tim\Desktop\NEDD60~1\TEST~1.DIS\test.exe] Error 1257
FATAL: Failed unexpectedly in Scons C backend compilation.
Nuitka:WARNING:     Complex topic! More information can be found at
Nuitka:WARNING: https://nuitka.net/info/scons-backend-failure.html
Nuitka-Reports: Compilation crash report written to file 'nuitka-crash-report.xml'.
```

`-low-memory`, `--lto=yes/no` were also tried but could not resolve the issue.

## Further Notes

This issie is similar to #3498.

--- a/nuitka/plugins/standard/ImplicitImports.py
+++ b/nuitka/plugins/standard/ImplicitImports.py
@@ -784,6 +784,7 @@ According to Yaml 'overridden-environment-variables' configuration."""
         "tensorflow.**.test_util",  # Not performance relevant.
         "google.protobuf.*_pb2",  #  Too large generated code
         "sqlfluff.dialects.*",  #  Too large generated code
+        "sqlalchemy.dialects.oracle.dictionary",  #  Too large generated code
     )
 
     def decideCompilation(self, module_name):

